src/Microsoft.OData.Core/Uri/NodeToStringBuilder.cs

```diff
@@ -470,7 +471,245 @@ internal String TranslateNode(QueryNode node)
+            {
+                leftIn = ExpressionConstants.It + ExpressionConstants.SymbolForwardSlash + leftIn;
+                translatedNode = leftIn + inOperator + rightIn;
+            }
+
+            return translatedNode;
+        }
+
+        /// <summary>
+        /// Bind $it to the <see cref="AnyNode"/> translated string.
+        /// </summary>
+        /// <param name="node">node to bind.</param>
+        /// <param name="filterClauseRangeVariable">The <see cref="FilterClause"/> range variable.</param>
+        /// <returns>The translated string with $it binding.</returns>
+        private string BindAnyNode(AnyNode node, ResourceRangeVariable filterClauseRangeVariable)
+        {
+            string translatedNode = this.TranslateNode(node);
+            ResourceRangeVariableReferenceNode leftRangeVariableNode = GetResourceRangeVariableReferenceNode(node);
+
+            const string slashAny = "/any";
+            string[] anySeparator = { slashAny };
+            string[] substrings = translatedNode.Trim().Split(anySeparator, StringSplitOptions.RemoveEmptyEntries);
+            string leftAnyNodeSubstring = substrings[0];
+            string rightAnyNodeSubstring = substrings[1];
+
+            if (leftRangeVariableNode != null && IsDifferentSource(filterClauseRangeVariable, leftRangeVariableNode))
+            {
+                leftAnyNodeSubstring = ExpressionConstants.It + ExpressionConstants.SymbolForwardSlash + leftAnyNodeSubstring;
+                translatedNode = leftAnyNodeSubstring + slashAny + rightAnyNodeSubstring;
+            }
+
+            return translatedNode;
+        }
+
+        /// <summary>
+        /// Bind $it to the <see cref="SingleValueFunctionCallNode"/> translated string.
+        /// </summary>
+        /// <param name="node">node to bind.</param>
+        /// <param name="filterClauseRangeVariable">The <see cref="FilterClause"/> range variable.</param>
+        /// <returns>The translated string with $it binding.</returns>
+        private string BindSingleValueFunctionCallNode(SingleValueFunctionCallNode node, ResourceRangeVariable filterClauseRangeVariable)
+        {
+            string translatedNode = this.TranslateNode(node);
+            ResourceRangeVariableReferenceNode firstParameterRangeVariableNode = GetResourceRangeVariableReferenceNode(node);
+
+            char[] separators = { '(', ')' };
+            string[] subtrings = translatedNode.Trim().Split(separators, StringSplitOptions.RemoveEmptyEntries);
+            string withinBrackets = subtrings[1];
+            char[] parameterSeparators = { ',' };
+            string[] parameters = withinBrackets.Trim().Split(parameterSeparators, StringSplitOptions.RemoveEmptyEntries);
+            string leftParameter = parameters[0];
+            string rightParameter = parameters.Length == 2 ? parameters[1] : String.Empty;

```

Comment:
all of this logic should be moved inside the conditional and only executed in the case that we are inserting the $it.